tests/instances.tcl

```diff
@@ -92,10 +91,30 @@ proc spawn_instance {type base_port count {conf {}}} {
         close $cfg
 
         # Finally exec it and remember the pid for later cleanup.
-        set pid [exec_instance $type $cfgfile]
-        lappend ::pids $pid
+        set retry 100
+        while {$retry} {
+            set pid [exec_instance $type $cfgfile]
+            after 1000
+
+            # Check availability
+            if {[server_is_up 127.0.0.1 $port 100] == 0} {
+                puts "Starting $type #$j at port $port failed, try another"
+                incr retry -1
+                set port [find_available_port [expr {$port+[randomInt 10]}]]
```

Comment:
this patch is based on old code, please rebase.
in Redis 6.0.4 (a2ae46352 and 1aee695e5) the `find_available_port` function was changed to take another argument.
i think the random isn't needed, just call it again and hope for the best.

Reply:
Sorry I don't notice a2ae463 and 1aee695, the random isn't needed, just attemp to choose the next port